src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy

```diff
@@ -16,31 +19,35 @@ import org.gradle.api.tasks.compile.JavaCompile
 import org.gradle.process.CommandLineArgumentProvider
 import org.gradle.util.VersionNumber
 
+import javax.inject.Inject
 import java.lang.reflect.Field
 
 /**
- * Changes Java annotation processor arguments to set the room.schemaLocation via a CommandLineArgsProcessor
+ * Changes annotation processor arguments to set the room.schemaLocation via a CommandLineArgsProcessor
  * which accurately sets it as an output of the task.  Note that it is common for people to set the value
  * using the defaultConfig in the android extension, which causes every compile task to use the same
  * output directory, causing overlapping outputs and breaking cacheability again.  To avoid this, we
  * change the location for each task to write to a task-specific directory, ensuring uniqueness across
  * compile tasks.  We then add a task to merge the generated schemas back to the originally
  * specified schema directory as a post-compile step.
  *
- * For kapt tasks, we cannot implement a CommandLineArgsProcessor, so we change the value to be relative
- * for the purposes of snapshotting inputs, then change it to the absolute path of a task-specific directory
- * right before the task executes.  We then merge it back to the originally specified directory using the
- * aforementioned merge task.  Note that this workaround only works with Kotlin Gradle plugin 1.3.70 or
- * higher.
+ * This workaround adds a 'room' project extension which can be used to configure the annotation processor:
+ *
+ * <pre>
+ *     room {
+ *         schemaLocationDir = file("roomSchemas")
```

Comment:
I think it would be good to also document this in the README.

Reply:
Yeah, I think you're right.  I've added a blurb about it - let me know what you think...